samza-api/src/main/java/org/apache/samza/operators/StreamGraphBuilder.java

```diff
@@ -27,6 +27,8 @@
  */
 @InterfaceStability.Unstable
 public interface StreamGraphBuilder {
+  static final String BUILDER_CLASS_CONFIG = "job.graph.builder.class";
```

Comment:
Can you please add some javadoc for what this config is for? We can later add the same to the config table.

I am also curious why this is called "job.graph.builder.class". Ideally, I would expect a graph to be composed of one or more jobs. But making it as a sub-namespace of job implies that a job itself contains a graph. This is based on my understanding of what we mean by a "job" today, in Samza. A job, today, indicates a single stage of processing defined by the user. Are we going to change the meaning of what a "job" means? 

I think applying all the stream graph related configurations "above" the job abstraction will simplify our configuration without impacting the notion of a job. My 2 cents :)